matcher_factories.go

```diff
@@ -249,13 +329,28 @@ func EqString(value string) string {
 	return ""
 }
 
+func NotEqString(value string) string {
+	RegisterMatcher(&NotEqMatcher{Value: value})
+	return ""
+}
+
 func AnyString() string {
-	RegisterMatcher(NewAnyMatcher(reflect.TypeOf((string)(""))))
+	RegisterMatcher(NewAnyMatcher(reflect.TypeOf("")))
 	return ""
 }
 
+func EqStringSlice(value []string) []string {
+	RegisterMatcher(&EqMatcher{Value: value})
+	return nil
+}
+
+func NotEqStringSlice(value []string) []string {
+	RegisterMatcher(&NotEqMatcher{Value: value})
+	return nil
+}
```

Comment:
These seemed to be missing so I added them for completeness. Let me know if they had been left out for a reason.

Reply:
I think I simply forgot them. 👍

Reply:
Turns out that there were way more factory methods missing in here so I added them all for completeness